File: pinyougou-cart-interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import com.pinyougou.pojogroup.Cart;

import java.util.List;

public interface CartService {

    //1 添加商品到购物车
    List<Cart> addGoodsToCartList(List<Cart> cartList, long itemId, int num);
    //2 从redis中获取购物车列表
    public List<Cart> findCartListFromRedis(String userName);
    //3 将购物车列表 保存到redis中
    public void saveCartListToRedis(String username,List<Cart> cartList);
    //4 合并cookie和redis购物车列表到redis中
    public List<Cart> mergeCartList(List<Cart> cookieCartList,List<Cart> redisCartList);
}

File: pinyougou-cart-web/src/main/webapp/js/service/cartService.js
app.service("cartService",function ($http) {
    //1.查询购物车列表
    this.findCartList = ()=>{
        return $http.get("./cart/findCartList.do");
    }
    //2.添加到购物车
    this.addGoodsToCartList=(itemId,num)=>{
        return $http.get("./cart/addGoodsToCartList.do?itemId="+itemId+"&num="+num);
    }
})